src/main/java/uk/nhs/hee/tis/trainee/sync/service/ConditionsOfJoiningSyncService.java

```diff
@@ -63,7 +65,12 @@ public void syncRecord(Record conditionsOfJoiningRecord) {
           = findById(conditionsOfJoining.getProgrammeMembershipUuid());
       savedCoj.ifPresent(
           ofJoining -> conditionsOfJoining.setSyncedAt(ofJoining.getSyncedAt()));
-      repository.save(conditionsOfJoining);
+
+      //saving the Conditions of Joining may set its syncedAt value
+      Record updatedCojRecord = mapper.toRecord(repository.save(conditionsOfJoining));
```

Comment:
This will be reworked to publish the sort of DTO that is needed by the notification service, i.e. an AggregateProgrammeMembershipDto (I think?)